clients/js/test/T22/deposit.test.ts

```diff
@@ -0,0 +1,201 @@
+import { appendTransactionMessageInstruction, pipe } from '@solana/web3.js';
+import {
+  createDefaultSolanaClient,
+  createDefaultTransaction,
+  createT22NftWithRoyalties,
+  expectCustomError,
+  generateKeyPairSignerWithSol,
+  signAndSendTransaction,
+} from '@tensor-foundation/test-helpers';
+import {
+  fetchWhitelistV2,
+  getUpdateWhitelistV2Instruction,
+  Mode,
+  operation,
+  TENSOR_WHITELIST_ERROR__FAILED_FVC_VERIFICATION,
+  TENSOR_WHITELIST_ERROR__FAILED_MERKLE_PROOF_VERIFICATION,
+  TENSOR_WHITELIST_ERROR__FAILED_VOC_VERIFICATION,
+} from '@tensor-foundation/whitelist';
+import test from 'ava';
+import {
+  CurveType,
+  fetchPool,
+  findNftDepositReceiptPda,
+  getDepositNftT22InstructionAsync,
+  PoolConfig,
+  PoolType,
+} from '../../src/index.js';
+import { createPool, createWhitelistV2 } from '../_common.js';
+
+test('deposit non-whitelisted NFT fails', async (t) => {
```

Comment:
Maybe 2 additional tests could fit in here:

1. it can deposit a whitelisted NFT
2. it can't deposit a whitelisted NFT with incorrect rem accs (transferHookAccs)

Reply:
Added